File: Week3/Group_get_TreeHeight.R
#Language: R
#Auther: Cong Liu (cong.liu20@imperial.ac.uk)
#Script: get_TreeHeight.R
#Work Path: CMEECourseWork/Week3
#Input: Data/trees.csv
#Function: Calculate heights of trees given distance of each tree 
#          from its base and angle to its top, using  the trigonometric formula 
#Output: Results/trees_treeheights.csv
#Usage: Rscript get_TreeHeight.R [Input]
#Date: Oct, 2020

args = commandArgs(trailingOnly = T)

data = read.csv(args, header = T)
a = nrow(data)

TreeHeight <- function(degrees, distance){
  radians <- degrees * pi / 180
  height <- distance * tan(radians)
  return (height)
}

Tree.Height.m = c()
for ( i in 1:a ){
  distance = as.numeric(data$Distance.m[i])
  degree = as.numeric(data$Angle.degrees[i])
  height = TreeHeight(degree, distance)
  Tree.Height.m[i] = height 
}

data_more = data.frame(data, Tree.Height.m)

inputfile = unlist(strsplit(args, split = "/"))[2]

inputfilename = unlist(strsplit(inputfile, split = ".csv"))[1]


e = paste("Results/", inputfilename, sep = "")
e = paste(e, "_treeheights.csv", sep = "")
write.csv(data_more, e, row.names = F)

File: Week3/Group_get_TreeHeight.py
"""Language: Python3
   Auther: Cong Liu (cong.liu20@imperial.ac.uk)
   Script: get_TreeHeight.py
   Work Path: CMEECourseWork/Week3
   Dependencies: sys, math
   Input: Data/trees.csv
   Function: Calculate heights of trees given distance of each tree 
             from its base and angle to its top, using  the trigonometric formula 
   Output: Results/trees_treeheights.csv
   Usage: python get_TreeHeight.py [Input]
   Date: Oct, 2020"""

import sys
import math

inpu = sys.argv[1]
file = open(inpu, "r")

data = []
for line in file:
    line = line.strip().split(",")
    data.append(line)
file.close()

def TreeHeight(distance, degree):
    """Calculate heights of trees.
       
       Input:
       distance: the distance from tree base
       degree: angle to its top
       
       Output: Height of tree"""
    pi = math.pi
    deg = degree*pi/180
    tanv = math.tan(deg)
    height = distance * tanv
    return height

res = []
data[0].append("Tree.Height.m")
for i in range(1,len(data)):
    height = TreeHeight(float(data[i][1]), float(data[i][2]))
    data[i].append(height)

result_name = "Results/" + inpu.split("/")[1].split(".")[0] + "_treeheights.csv"
result = open(result_name, "w")
for i in data:
    result.write(i[0]+","+i[1]+","+i[2]+","+str(i[3]))
    result.write("\n")

result.close() 
